client/jetpack-connect/authorize.js

```diff
@@ -681,7 +681,7 @@ export class JetpackAuthorize extends Component {
 					eventName="calypso_jpc_authorize_chat_initiated"
 					label={ helpButtonLabel }
 				>
-					<HelpButton label={ helpButtonLabel } url={ wooDna.helpUrl } />
+					<HelpButton label={ helpButtonLabel } url={ wooDna.getServiceHelpUrl() } />
```

Comment:
1. Service help url might be empty. Should we check if it exists before rendering help button?
2. The url is taken from query params. Does it make sense to sanitize it for security reasons? E.g. inside wooDnaConfig. As I understand nothing blocks us from passing `javascript:alert('Boom!')` as service help url param.

Reply:
> Service help url might be empty. Should we check if it exists before rendering help button?

At this point, we are in a `WooDNA` flow (there's a check at the start of the function). Which means that the help URL *should* exist. If if doesn't, then it will default to `https://jetpack.com/contact-support` ([see here](https://github.com/Automattic/wp-calypso/blob/fcdc014fba91ba6b601412469f21bcb21b87f298/client/jetpack-connect/help-button.jsx#L27)), which is better than not having the help button at all.

> The url is taken from query params. Does it make sense to sanitize it for security reasons? E.g. inside wooDnaConfig. As I understand nothing blocks us from passing javascript:alert('Boom!') as service help url param.

I'm not a big fan of "defensive programming". That URL comes directly from the wp.com server, which we control. If an attacker has access to modify the URL, we're in much bigger trouble.

That said, I just updated the server diff to remove the `woodna_*` URL parameters if the current flow is not WooDNA. That prevents a malicious plugin from setting them up arbitrarily. Thanks for making me think about security! :)

Reply:
👍 I like taken precautions, I agree they are good enough, thanks for clarifying.